examples/jn516x/tsch/uart1-test-node/uart1-test-node.c

```diff
@@ -0,0 +1,255 @@
+
+  memset(rx_buf, '\0', sizeof(rx_buf));
+  /* Define process that handles data */
+  process_start(&rx_data_process ,NULL);
+  /* Define ringbuffer control */
+//  ringbufindex_init(&ringbuf,RX_BUFFER_SIZE);
+  /* Initialise UART1 */
+  uart1_init(UART1_BAUD_RATE); 
+  /* Callback received byte */
+  uart1_set_input(handleRxChar);
+
+  /* Start RICH stack */
+  if(is_coordinator) {
+    uip_ipaddr_t prefix;
+    uip_ip6addr(&prefix, 0xaaaa, 0, 0, 0, 0, 0, 0, 0);
+    rich_init(&prefix);
+  } else {
+    rich_init(NULL);
+  }
+  printf("Starting RPL node\n");
+  
+  rest_init_engine();
+  rest_activate_resource(&resource_coap_rx_uart1, "UART1-RX");
+  rest_activate_resource(&resource_coap_tx_uart1, "UART1-TX");
+
+  PROCESS_END();
+}
+
+
+/*---------------------------------------------------------------------------*/
+PROCESS_THREAD(rx_data_process, ev, data)
+{
+  PROCESS_BEGIN();
+  
+  /* Process is polled whenever data is available from uart isr */
+  uint8_t c;
+
+  while(1) {
+    PROCESS_YIELD_UNTIL(ev == PROCESS_EVENT_POLL);
+    /* Read RX ringbuffer. ASCII chars Output when LF is seen. 
+       If overflowed, strings are skipped */ 
+    do {
+      if (get_ringbuf(&c) == -1) {
+        break;    /* No more rx char's in ringbuffer */
+      } else {
+        if (rx_buf_index == RX_BUFFER_SIZE) {   /* Skip current content if buffer full */
+          rx_buf_index = 0;
+        } 
+        rx_buf[rx_buf_index++] = c;
+        if ((c == '\n')||(c == '\r')) {
+          rx_buf[rx_buf_index] = '\0';
+          printf("RX on UART1: %s", rx_buf);   
+          /* Signal event to coap clients.
+             Demo assumes data is consumed before new data comes in */
+          event_coap_rx_uart1_handler();     
+          rx_buf_index = 0;      
+        }
+      }
+    } while (1);
+  }
+  PROCESS_END();
+}
+
+/*************************************************************************/
+/* Local test functions                                                  */
+/*************************************************************************/
+/* TX function for UART1 */
+static void string2uart1(uint8_t *c)
+{
+  while (*c!= '\0') {
+    uart1_writeb(*c);
+    c++;
+  }
+}
+
+/* handleRxChar runs on uart isr */
+static int handleRxChar(uint8_t c)
+{
+  if (put_ringbuf(c) == -1) {
+    printf("Ringbuf full. Skip char\n");
+  } else {
+    /* Send event to rx data handler */
+    process_poll(&rx_data_process);
+  }
+  return 0;
+}
+
+/* Simple ringbuffer
+   if tail==head, no data has been written yet on that position. So, empty buffer
+   is also initial state */
+static int get_ringbuf(uint8_t *c)
+{
+  int return_val = 0;
+
+  uart1_disable_interrupts();
+  if (head_index != ((tail_index + 1)&RINGBUF_MAX_INDEX)) {
+    tail_index = ((tail_index + 1)& RINGBUF_MAX_INDEX);
+    *c = ringbuf[tail_index];
+  } else {
+    return_val = -1;
+  }
+  uart1_enable_interrupts();
+  return return_val;
+}
+
+static int put_ringbuf(uint8_t c)
```

Comment:
code style: `static int` should be alone on a first line. I saw a few more instances above